python/ray/train/torch/xla/config.py

```diff
@@ -0,0 +1,126 @@
+import os
+import uuid
+from dataclasses import dataclass
+
+import ray
+from ray._private.ray_constants import NEURON_CORES
+from ray.train import BackendConfig
+from ray.train._internal.utils import get_address_and_port
+from ray.train._internal.worker_group import WorkerGroup
+from ray.train.backend import Backend
+from ray.util import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+@dataclass
+class TorchXLAConfig(BackendConfig):
+    """
+    Configuration for torch XLA setup.
+    See https://pytorch.org/xla/release/1.13/index.html for more info.
+
+    Args:
+        runtime: Runtime to use for training. Supported values are "xrt", "pjrt".
+            Currently, only "xrt" is supported.
+        accelerator_type: The accelerator type used to differentiate the XLA backend.
+            Currently, only "neuron_cores" is supported.
+
+    """
+
+    runtime: str = "xrt"
+    accelerator_type: str = NEURON_CORES
+
+    @property
+    def backend_cls(self):
+        if self.accelerator_type == NEURON_CORES:
+            return _TorchAwsNeuronXLABackend
+        else:
+            raise NotImplementedError
+
+
+def _kill_xrt_server():
+    import subprocess
+
+    subprocess.call(["pkill", "-f", "xrt_run_server"])
+
+
+def _set_xla_env_vars():
+    # https://pytorch.org/docs/1.13/elastic/run.html#environment-variables
+    context = ray.train.get_context()
+
+    os.environ["LOCAL_RANK"] = str(context.get_local_rank())
+    os.environ["RANK"] = str(context.get_world_rank())
+    os.environ["LOCAL_WORLD_SIZE"] = str(context.get_local_world_size())
+    os.environ["WORLD_SIZE"] = str(context.get_world_size())
+    os.environ["GROUP_RANK"] = str(context.get_node_rank())
+    os.environ["GROUP_WORLD_SIZE"] = str(
+        context.get_world_size() / context.get_local_world_size()
+    )
```

Comment:
I haven't seen any better documentation to be honest and my knowledge is slim in this area.

Also, do we have a way to pass addition work/actions before the training starts without extending the class? For example, I want to add a new EFA configuration as env_var for this train/run. More like a function to `on_start` which will be executed on all workers.

Reply:
Okay can we just add a TODO here. If we need to support heterogenous cluster we can add more robust logic in `on_training_start` to fetch the topology of the cluster and then set it.

For the question on `on_start`, just capturing here the conversation we had offline - this can be defined by the user in in `train_loop_per_worker`.